File: java/polimorfism_java/Esfera.java
public class Esfera extends Figura {

    public Esfera(String nome, float raio) {
        super(nome, raio);
    }

    @Override
    public void display() {
        System.out.println("nome: "+getNome());
        System.out.println("raio: "+getDim1());
        System.out.println("area: "+getArea());
        System.out.println("volume: "+getVolume());
    }

    @Override
    public float getArea() {return (float)(4.0*Math.PI*getDim1()*getDim1());}

    @Override
    public float getVolume() {return (float)((4.0/3.0)*Math.PI*Math.pow(getDim1(),3));}
}

File: exercises/polimorfism/Esfera.cpp
#include "Esfera.h"
#include <iostream>
#include <cmath>

using namespace std;

Esfera::Esfera(std::string nome, float raio)
    : Figura(nome, raio) {}

void Esfera::display() {
    cout<<"nome: "<<getNome()<<endl;
    cout<<"raio: "<<getDim1()<<endl;
    cout<<"area: "<<getArea()<<endl;
    cout<<"volume: "<<getVolume()<<endl;
}

float Esfera::getArea() const {
    return 4.0f*M_PI*getDim1()*getDim1();
}

float Esfera::getVolume() const {
    return (4.0f/3.0f)*M_PI*getDim1()*getDim1()*getDim1();